tools/devops/automation/templates/build/build.yml

```diff
@@ -516,3 +516,14 @@ steps:
 - template: ../common/teardown.yml
   parameters:
     keyringPass: ${{ parameters.keyringPass }}
+
+- pwsh: |
+    Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY\xamarin-macios\tools\devops\automation\scripts\GitHub.psm1
+    New-GitHubComment -Header "Build failed" -Emoji ":fire:" -Description "Build failed for the job '$Env:SYSTEM_JOBDISPLAYNAME'"
+  condition: failed()
+  displayName: 'Report build failure'
+  env:
+    BUILD_REVISION: $(Build.SourceVersion)
+    CONTEXT: "Build"
+    GITHUB_TOKEN: $(GitHub.Token)
+    SYSTEM_ACCESSTOKEN: $(System.AccessToken)
```

Comment:
This is going to happen only if the previous step fails, meaning that the team down failed. Which is incorrect. If you want to ensure that it is when the build fails, you need to move it AFTER the build step here: https://github.com/xamarin/xamarin-macios/blob/main/tools/devops/automation/templates/build/build.yml#L323 Otherwise this would be misleading, since another step could have triggered this condition. For example an step that should download and did not.

Reply:
The point is to leave a GitHub comment if anything in this file fails - since in that case we currently don't report anything, you have to go back (after at least half a day when you realize nothing happened) and check the result.

From what I understand any step with `continueOnError: true` won't trigger this, so only fatal steps would (and that's exactly the behavior I want).

I could change the message to something like `"Something in the 'build' pipeline failed"` if that's better?